config/initializers/renderers.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+ActionController::Renderers.add :csv do |obj, options|
+  filename = options[:filename] || 'data'
+  str = obj.respond_to?(:to_csv) ? obj.to_csv : obj.to_s
+  send_data str, type: Mime[:csv], disposition: "attachment; filename=#{filename.chomp('.csv')}.csv"
```

Comment:
@frothedoatmilk looks like this is what you needed when you went on your ft pdf crusade